main/src/ca/uwaterloo/flix/language/phase/Typer.scala

```diff
@@ -321,6 +321,7 @@ object Typer {
               Scheme.checkLessThanEqual(inferredSc, declaredScheme, classEnv) match {
                 // Case 1: no errors, continue
                 case Validation.Success(_) => // noop
+                case Validation.SuccessWithFailures(_, _) => ???
                 case Validation.Failure(errs) =>
```

Comment:
I think -- for Validation -- this could be one of the rare cases where we use an OR pattern (if it works).

Can you write:

```scala
Validation.SuccessWithFailures(_, errs) | Validation.Failure(errs) => 
```

Or does that not work?

Reply:
Alternatively, we could always write:

```
case Validation.Success(...) =>
case failure => ... failure.errors ...
```

maybe that is cleaner?

Reply:
Throughout we should consider `SuccessWithFailures` as `Failure`.

The only exception is a specific point inside the language server -- which I will identify.